### PR TITLE
[RLlib] Upgrade gym version to 0.21 and deprecate pendulum-v0.

### LINKED ISSUE #17827: Testing that the newest version of Gym doesn't break anything in RLlib befor I release it
Hey,

I'm about to cut a release of Gym based on what's currently on the master branch for it. Joseph Suarez (Neural MMO) said I should ask you guys to run all your RLlib tests with this to make sure nothing is broken.

@sven1977

### PR DESCRIPTION
**Pull request by avnishn**
Many tests in rllib depended on pendulum v0,
however in gym 0.21, pendulum v0 was deprecated
in favor of pendulum v1. This may change reward
thresholds, so will have to potentially rerun
all of the pendulum v1 benchmarks, or use another
environment in favor.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — COMMENTED** [review]
This is awesome!!!

how different is v1 from v0? do you think it's necessary to re-benchmark, adjust all the threshold etc?

also, I am seeing a lot of CI errors like:
gym.error.DeprecatedEnv: Env Pendulum-v1 not found (valid versions include ['Pendulum-v0'])
probably need to un-pin some other places?

**Review by sven1977 — APPROVED** [review]
This looks great! Thanks for this difficult PR, dealing with all these breaking changes @avnishn !

Just a few notes for future PRs:
- When you add new test files, make sure you activate them in our ray.rllib.BUILD file, otherwise, they won't be included in the CI testing.
- When writing new functions/classes and their docstrings, let's make sure we only define type-hints (incl. return type) in the function's signature and not anymore in the docstring's "Args" and "Returns". I know we have to retro-fit more old code to fit this pattern.
- When changing existing (non-private) function signatures, please make sure the old args are still supported (soft-deprecate them first via the tools in ray.rllib.utils.deprecation). For an example, see `ray.rllib.utils.numpy.convert_to_numpy()`.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -578,19 +578,15 @@
...
     - bazel test --config=ci $(./scripts/bazel_export_options) --test_tag_filters=-example,-flaky,-py37,-soft_imports,-gpu_only python/ray/tune/...
     - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=example,-tf,-pytorch,-py37,-flaky,-soft_imports,-gpu_only python/ray/tune/...
 
 - label: ":octopus: Tune tests and examples {2/2}"
   conditions: ["RAY_CI_TUNE_AFFECTED"]
   commands:
     - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/travis/upload_build_info.sh; fi }; trap cleanup EXIT
-    - TUNE_TESTING=1 PYTHON=3.7 ./ci/travis/install-dependencies.sh
-    # Because Python version changed, we need to re-install Ray here
-    - rm -rf ./python/ray/thirdparty_files; rm -rf ./python/ray/pickle5_files; ./ci/travis/ci.sh build
```
Comment (richardliaw) [inline .buildkite/pipeline.yml:593]: Why did we commit this change?

File: docker/ray-ml/Dockerfile [hunk 2]
```diff
@@ -37,4 +37,4 @@ RUN sudo apt-get update \
 RUN python -c "import tensorflow_probability"
 
 # Install Atari ROMs. Previously these have been shipped with atari_py \
-RUN ./install_atari_roms.sh
+# RUN ./install_atari_roms.sh
```
Comment (richardliaw) [inline docker/ray-ml/Dockerfile:40]: Can we clean this up?

File: ci/travis/install-dependencies.sh [hunk 3]
```diff
@@ -358,12 +358,12 @@ install_dependencies() {
...
-      bash "${WORKSPACE_DIR}"/rllib/utils/install_atari_roms.sh
-    else
-      echo "Not installing Atari roms on ${OSTYPE}"
-    fi
+    # # Install Atari ROMs. Previously these have been shipped with atari_py
+    # if [[ "${OSTYPE}" = linux* ]]; then
+    #   bash "${WORKSPACE_DIR}"/rllib/utils/install_atari_roms.sh
+    # else
+    #   echo "Not installing Atari roms on ${OSTYPE}"
+    # fi
```
Comment (richardliaw) [inline ci/travis/install-dependencies.sh:366]: Can we clean this up too?

**Pull request merged**